### PR TITLE
feat[v2]: generate schemas for v3 config versions

### PR DESCRIPTION
**Pull request by nkubala**
This generates JSON schemas for all new-style (`v3`) config versions.

* Fix up comments for schemas to play nice with our validator
* Fix up a small bad merge
* Skip `TestUpgradeToNextVersion()` for the latest v2 schema since we don't support `skaffold fix` between old- and new-style configs yet

cc @yuwenma

### CODE REVIEW

**Comment by aaron-prindle** [comment]
seems this doesn't meet conventional commits check for the commit message atm

File: pkg/skaffold/schema/latest/v2/config.go [hunk 1]
```diff
@@ -571,7 +579,7 @@ type KptV2Deploy struct {
 	// InventoryNamespace *alpha* sets the inventory namespace.
 	InventoryNamespace string `yaml:"namespace,omitempty"`
 
-	// Forces is used in `kpt live init`, which forces the inventory values to be updated, even if they are already set.
+	// Force is used in `kpt live init`, which forces the inventory values to be updated, even if they are already set.
 	Force bool `yaml:"false,omitempty"`
```
Comment (yuwenma) [inline pkg/skaffold/schema/latest/v2/config.go:583]: correct: "false" -->"force"

**Review by yuwenma — APPROVED** [review]
LGTM

**Review by SSGoku369 — COMMENTED** [review]
ok

**Comment by SSGoku369** [comment]
ok

**Comment by nkubala** [comment]
closing this and reopening against a newer branch

**Pull request closed without merging**